tequilapi/endpoints/proposals.go

```diff
@@ -85,21 +89,23 @@ func proposalToRes(p dto_discovery.ServiceProposal) proposalRes {
 func mapProposalsToRes(
 	proposalArry []dto_discovery.ServiceProposal,
 	f func(dto_discovery.ServiceProposal) proposalRes,
+	quality func(proposalRes) proposalRes,
 ) []proposalRes {
 	proposalsResArry := make([]proposalRes, len(proposalArry))
 	for i, proposal := range proposalArry {
-		proposalsResArry[i] = f(proposal)
+		proposalsResArry[i] = quality(f(proposal))
 	}
 	return proposalsResArry
 }
 
 type proposalsEndpoint struct {
-	mysteriumClient server.Client
+	mysteriumClient      server.Client
+	mysteriumMorqaClient server.MorqaClient
```

Comment:
It should be `QualityOracle` interface, and user can You our Morqa service, but can use another quality provider also